piopio/piopio_be/views.py

```diff
@@ -299,21 +324,55 @@ def search(self, request):
             posts_queryset = self.queryset
 
             for content in contents:
-                posts_queryset = posts_queryset.filter(content__icontains=content)
+                posts_queryset = posts_queryset.filter(
+                    content__icontains=content)
 
         except ValueError:
             return Response({'content': 'Not specified'}, status.HTTP_404_NOT_FOUND)
 
         posts_queryset = add_likes_and_retweets(posts_queryset, request.user)
         page = self.paginate_queryset(posts_queryset)
-        serialized_posts = serializers.PostSerializerWLikedRetweet(page, many=True)
+        serialized_posts = serializers.PostSerializerWLikedRetweet(
+            page, many=True)
         return self.get_paginated_response(serialized_posts.data)
 
+    @action(methods=['GET'], detail=False, url_path="details/(?P<postpk>[^/.]+)", url_name="detail_post")
+    def details(self, request, postpk):
+        try:
+            post = self.get_queryset().filter(id=postpk)
+            parent = post.get().parent.all()
+            child = self.get_queryset().filter(parent__id__exact=postpk)
+            detail = list(chain(parent, post, child))
```

Comment:
Union of querysets without casting to list

Reply:
Changed it to union.